sdks/python/apache_beam/io/mongodbio.py

```diff
@@ -241,6 +275,27 @@ def _get_split_keys(self, desired_chunk_size_in_mb, start_pos, end_pos):
               max={'_id': end_pos},
               maxChunkSize=desired_chunk_size_in_mb)['splitKeys'])
 
+  def _get_buckets(self, desired_chunk_size, start_pos, end_pos):
+    if start_pos >= end_pos:
+      # single document not splittable
+      return []
+    size = self.estimate_size()
+    bucket_count = size // desired_chunk_size
```

Comment:
The split function will likely be called recursively for dynamic rebalancing, so for a range with start_pos and end_pos, it can be further split upon backend request, so it might not reasonable to always use the total collection size divide by desired_chunk_size to calculate the bucket count. Is it possible to only get the buckets within the give _id range? and we can probably use an average document size times the number of documents to calculate the size of the range being split.

Reply:
Thanks, @y1chi!
I will look closer how it works in Java sdk and will try to filter by _id ranges.